lib/SPIRV/OCL20ToSPIRV.cpp

```diff
@@ -1027,9 +1027,23 @@ void OCL20ToSPIRV::transBuiltin(CallInst *CI, OCLBuiltinTransInfo &Info) {
   unsigned ExtOp = ~0U;
   if (StringRef(Info.UniqName).startswith(kSPIRVName::Prefix))
     return;
-  if (OCLSPIRVBuiltinMap::find(Info.UniqName, &OC))
-    Info.UniqName = getSPIRVFuncName(OC);
-  else if ((ExtOp = getExtOp(Info.MangledName, Info.UniqName)) != ~0U)
+  if (OCLSPIRVBuiltinMap::find(Info.UniqName, &OC)) {
+    if (OC == OpImageRead) {
+      // There are several read_image* functions defined by OpenCL C spec, but
+      // all of them use the same SPIR-V Instruction - some of them might only
+      // differ by return type, so, we need to include return type into the
```

Comment:
Including the return type in the mangled name does not sound like the correct way to handle this issue.

SPIR-V 1.4 adds the SignExtend and ZeroExtend image operands.  It would be better to make use of those somehow.

We have a downstream patch that uses a single bit in the image operands to disambiguate.  I could try to upstream that instead if you are interested, but it will require a bit of rework and it will not be conformant to any pre-1.4 spec.

Reply:
We also should disambiguate between the following 2 function calls:
```
uint4 c = read_imageui(input, (int4)(0, 0, 0, 0));
float4 f = read_imagef(input, (int4)(0, 0, 0, 0));
```
I don't think the image operands can help to represent these calls in SPIR-V-friendly LLVM IR.

Reply:
Ah sorry, I didn't realize this was only about the SPIR-V representation in LLVM IR.

LGTM then!